NitroxClient/GameLogic/Vehicles.cs

```diff
@@ -137,23 +137,59 @@ private void OnVehiclePrefabLoaded(TechType techType, GameObject prefab, NitroxI
         {
             // Partially copied from SubConsoleCommand.OnSubPrefabLoaded
             GameObject gameObject = Utils.SpawnPrefabAt(prefab, null, spawnPosition);
+
+            if (!gameObject)
+            {
+                Log.Error($"Failed to create gameObject from prefab {prefab.ToString()} for {id}");
+                return;
+            }
+
             gameObject.transform.rotation = spawnRotation;
             gameObject.SetActive(true);
             gameObject.SendMessage("StartConstruction", SendMessageOptions.DontRequireReceiver);
             CrafterLogic.NotifyCraftEnd(gameObject, CraftData.GetTechType(gameObject));
+            
             Rigidbody rigidBody = gameObject.GetComponent<Rigidbody>();
+            if (!rigidBody)
+            {
+                Log.Error($"Failed to create rigidBody from prefab {prefab.ToString()} for {id}");
+                return;
+            }
+
             rigidBody.isKinematic = false;
             NitroxEntity.SetNewId(gameObject, id);
+
             // Updates names and colours with persisted data
             if (techType == TechType.Seamoth || techType == TechType.Exosuit)
             {
                 Vehicle vehicle = gameObject.GetComponent<Vehicle>();
+                if (!vehicle)
+                {
+                    Log.Error($"Failed to get vehicle component from prefab {prefab.ToString()} for {id}");
+                    return;
+                }
+
                 if (dockingBayId.HasValue)
                 {
                     GameObject dockingBayBase = NitroxEntity.RequireObjectFrom(dockingBayId.Value);
+                    if (!dockingBayBase)
```

Comment:
Some of these are using RequireObjectFrom which will throw if the object doesn't exist.